src/averageAges.js

```diff
@@ -14,12 +14,18 @@
  *
  * @return {number}
  */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+function calculateMenAverageAge(people, century = 0) {
+  const male = people.filter(person => person.sex === 'm');
+  const peopleOfCentury = male.filter(person =>
+    Math.ceil(person.died / 100) === century);
+
+  const men = century ? peopleOfCentury : male;
+
+  const age = men.map(date => date.died - date.born);
+  const averageAge = Math.round(age.reduce((sumAge, personAge) =>
+    sumAge + personAge, 0) * 100 / age.length) / 100;
```

Comment:
you use same logic several times. So please move this logic to some `util function`